docs/user-guide/petset.md

```diff
@@ -0,0 +1,370 @@
+
+
+## Properties of a Pet
+
+The following examples demonstrate the properties of Pets.
+
+### Identity
+
+A minimal Pet Set might look like:
+
+{% include code.html language="yaml" file="petset.yaml" ghlink="/docs/user-guide/petset.yaml" %}
+
+For completeness it includes the `Service` definition. Lets create the Pet Set:
+
+```shell
+$ kubectl create -f petset.yaml
+service "nginx" created
+petset "nginx" deleted
+```
+
+The identity of a Pet maps to the following:
+
+#### Ordinal index
+you should see 2 pods with predictable names formatted thus: `$(petset name)-$(ordinal index assigned by petset controller)`
+
+```shell
+$ kubectl get po
+NAME      READY     STATUS    RESTARTS   AGE
+web-0     1/1       Running   0          10m
+web-1     1/1       Running   0          10m
+```
+
+#### Stable storage
+2 persistent volumes, one per pod. This is auto created by the Pet Set based on the `volumeTemplate` field
+
+```shell
+$ kubectl get pv
+NAME                                       CAPACITY   ACCESSMODES   STATUS    CLAIM               REASON    AGE
+pvc-90234946-3717-11e6-a46e-42010af00002   1Gi        RWO           Bound     default/www-web-0             11m
+pvc-902733c2-3717-11e6-a46e-42010af00002   1Gi        RWO           Bound     default/www-web-1             11m
+```
+
+#### Network identity
+The network identity has 2 parts.
+
+First, we created a headless Service that controls the domain within which we create Pets. The domain managed by this Service takes the form: `$(service name).$(namespace).svc.cluster.local`, where "cluster.local" is the [cluster domain]().
+
+```shell
+$ kubectl get svc
+NAME          CLUSTER-IP     EXTERNAL-IP       PORT(S)   AGE
+nginx         None           <none>            80/TCP    12m
+```
+
+As each pet is created, it gets a matching DNS name carved out of the parent domain, taking the form: `$(petname).$(governing service suffix)`, where the governing service is defined by the `serviceName` field on the Pet Set. Lets verify this assertion with a simple test. The containers are running nginx webservers, which by default will look for an index.html file in `/usr/share/nginx/html/index.html`. That directory is backed by a PersistentVolume created by the Pet Set. So lets write our hostname there (remember the Pet Set gives us a stable hostname):
+
+```shell
+for i in 0 1; do
+  kubectl exec web-$i -- sh -c 'echo $(hostname) > /usr/share/nginx/html/index.html';
+done
+```
+
+Now delete all pods in the petset:
+
+```shell
+$ kubectl delete po -l app=nginx
+pod "web-0" deleted
+pod "web-1" deleted
+```
+
+Wait for them to come back up, and try to retrieve the previously written hostname through the DNS name of the peer (remember the Pet Set also gives us stable storage, and that the hostname is linked to the DNS name).
```

Comment:
Might be worth letting users know (as the scheduler may schedule the pods anywhere) that whether the pods are scheduled on the same node or not, their storage will continue to be associated with the running pod.